src/Compilers/CSharp/Test/Semantic/Semantics/NonTrailingNamedArgumentsTests.cs

```diff
@@ -0,0 +1,646 @@
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+            var nodes = tree.GetCompilationUnitRoot().DescendantNodes();
+            var invocation = nodes.OfType<InvocationExpressionSyntax>().Single();
+            Assert.Equal("M(x: 1, x: 2, 3)", invocation.ToString());
+            Assert.Null(model.GetSymbolInfo(invocation).Symbol);
+        }
+
+        [Fact]
+        public void TestNamedParams3()
+        {
+            var source = @"
+class C
+{
+    static void M(int x, params int[] y)
+    {
+    }
+    static void Main()
+    {
+        M(y: 1, 2);
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.RegularLatest);
+            comp.VerifyDiagnostics(
+                // (9,11): error CS8321: Named argument 'y' is used out-of-position but is followed by an unnamed argument
+                //         M(y: 1, 2);
+                Diagnostic(ErrorCode.ERR_BadNonTrailingNamedArgument, "y").WithArguments("y").WithLocation(9, 11)
+                );
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+            var nodes = tree.GetCompilationUnitRoot().DescendantNodes();
+            var invocation = nodes.OfType<InvocationExpressionSyntax>().Single();
+            Assert.Equal("M(y: 1, 2)", invocation.ToString());
+            Assert.Null(model.GetSymbolInfo(invocation).Symbol);
+        }
+
+        [Fact]
+        public void TestNamedParams4()
+        {
+            var source = @"
+class C
+{
+    static void M(int x, params int[] y)
+    {
+    }
+    static void Main()
+    {
+        M(x: 1, y: 2, 3);
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.RegularLatest);
+            comp.VerifyDiagnostics(
+                // (9,9): error CS1501: No overload for method 'M' takes 3 arguments
+                //         M(x: 1, y: 2, 3);
+                Diagnostic(ErrorCode.ERR_BadArgCount, "M").WithArguments("M", "3").WithLocation(9, 9)
+                );
+        }
+
+        [Fact]
+        public void TestNamedInvalidParams()
+        {
+            var source = @"
+class C
+{
+    static void M(params int[] x, int y)
+    {
+    }
+    static void Main()
+    {
+        M(x: 1, 2);
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.RegularLatest);
+            comp.VerifyDiagnostics(
+                // (4,19): error CS0231: A params parameter must be the last parameter in a formal parameter list
+                //     static void M(params int[] x, int y)
+                Diagnostic(ErrorCode.ERR_ParamsLast, "params int[] x").WithLocation(4, 19),
+                // (9,14): error CS1503: Argument 1: cannot convert from 'int' to 'int'
+                //         M(x: 1, 2);
+                Diagnostic(ErrorCode.ERR_BadArgType, "1").WithArguments("1", "int", "int").WithLocation(9, 14)
```

Comment:
This error will change after rebasing. This bad diagnostic was fixed by https://github.com/dotnet/roslyn/pull/18923